### PR TITLE
Add a full list of pruning papers and their codes at ICML 2023 (10 in total)

### PR DESCRIPTION
**Pull request by sdc17**
Hi, @he-y, very thanks for your great list from which I learned a lot!

I am also working on pruning and I would like to contribute the full list of pruning papers and their codes at ICML 2023 (10 in total) as follows:

1.  [UPop: Unified and Progressive Pruning for Compressing Vision-Language Transformers](https://proceedings.mlr.press/v202/shi23e.html) | F | [PyTorch(Author)](https://github.com/sdc17/UPop)
2. [Gradient-Free Structured Pruning with Unlabeled Data](https://proceedings.mlr.press/v202/nova23a.html) | F | 
3. [Reconstructive Neuron Pruning for Backdoor Defense](https://proceedings.mlr.press/v202/li23v.html) | F | [PyTorch(Author)](https://github.com/bboylyg/RNP)
4. [UPSCALE: Unconstrained Channel Pruning](https://proceedings.mlr.press/v202/wan23a.html) | F | [PyTorch(Author)](https://github.com/apple/ml-upscale)
5. [SparseGPT: Massive Language Models Can be Accurately Pruned in One-Shot](https://proceedings.mlr.press/v202/frantar23a.html) | W | [PyTorch(Author)](https://github.com/IST-DASLab/sparsegpt)
6. [Why Random Pruning Is All We Need to Start Sparse](https://proceedings.mlr.press/v202/gadhikar23a.html) | W | [PyTorch(Author)](https://github.com/RelationalML/sparse_to_sparse)
7. [Fast as CHITA: Neural Network Pruning with Combinatorial Optimization](https://proceedings.mlr.press/v202/benbaki23a.html) | W | [PyTorch(Author)](https://github.com/mazumder-lab/CHITA)
8. [A Three-regime Model of Network Pruning](https://proceedings.mlr.press/v202/zhou23p.html) | W | [PyTorch(Author)](https://github.com/YefanZhou/ThreeRegimePruning)
9. [Instant Soup: Cheap Pruning Ensembles in A Single Pass Can Draw Lottery Tickets from Large Models](https://proceedings.mlr.press/v202/jaiswal23b.html) | W | [PyTorch(Author)](https://github.com/VITA-Group/instant_soup)
10. [Pruning via Sparsity-indexed ODE: a Continuous Sparsity Viewpoint](https://proceedings.mlr.press/v202/mo23c/mo23c.pdf) | W | [PyTorch(Author)](https://github.com/mzf666/sparsity-indexed-ode)

Could you please review this pr?  Please let me know if there is any further information I should provide for a final merge.

### CODE REVIEW

**Comment by cyfwry** [comment]
您好，您发给我的邮件已收到！
